### PR TITLE
feat: new links model utilities

### PR DESCRIPTION
**Pull request by OmarAlJarrah**
# Situation
The current SDK REST module lacks proper utilities for handling and parsing URL paths and query parameters from [HATEOAS](https://en.wikipedia.org/wiki/HATEOAS) links. This limitation prevents implementing robust link-following capabilities, a core principle of REST architectural style. Currently, when receiving links in API responses, we cannot efficiently extract path parameters, parse query parameters with different formatting strategies (CSV, SSV, pipes), or convert string values to appropriate data types. This makes it difficult to follow links and perform subsequent API operations.

# Task
- Develop a comprehensive URL path template parsing utility capable of extracting named parameters from path templates
- Create a query string parser supporting multiple collection format strategies (CSV, SSV, pipes, multi)
- Implement type conversion utilities with proper error handling for string-to-type conversion
- Refactor existing URL query parameter utilities for better organization and maintainability
- Create code generation templates for link builder methods to automate implementation in generated clients

# Action
1. Created new `UrlUtils.kt` with three core utilities:
   - `parseUrlPathTemplateParams`: Uses a segment-by-segment comparison algorithm to match template paths against actual paths, extracting parameter values from segments enclosed in `{` and `}` brackets
   - `parseUrlQueryParamsFromQuery`: Parses URL-encoded query strings, applying configurable destringification strategies for different collection formats
   - `resolveUrlQueryParamValuesType`: Applies type conversion using Jackson's type system, handling both single values and collections

2. Refactored `UrlQueryParamUtils.kt` with a cleaner design:
   - Created `UrlQueryParamStringifiers` class with companion object for stateless stringifier functions
   - Implemented `DestringifyQueryParam` functional interface for the reverse operation
   - Added `UrlQueryParamDestringifiers` with implementations for CSV, SSV, pipes formats
   - Created lookup map `swaggerCollectionFormatDestringifier` for format-based strategy selection
   - Ensured backward compatibility with existing code

3. Implemented `link_builder_method.mustache` template with:
   - Intelligent URL decomposition into path and query components
   - Dynamic path parameter extraction and type conversion
   - Query parameter extraction with format-specific destringification
   - Null safety and empty value handling throughout

4. Renamed parameter in `resolveUrlQueryParamValuesType` from `jacksonTypeRef` to `typeRef` for better naming consistency across the codebase

# Testing
- Implemented 380+ lines of comprehensive tests in `UrlUtilsTest.kt` with nested test classes:
  - `ParseUrlPathTemplateParamsTest`: Tests path parsing with various template patterns, including malformed templates and edge cases
  - `ParseUrlQueryParamsFromQueryTest`: Tests query parsing with different collection formats, encoding issues, and empty values
  - `ResolveUrlQueryParamValuesTypeTest`: Tests type conversion for primitives, collections, and handling of conversion failures

- Added additional tests for refactored `UrlQueryParamUtils`:
  - Verified all destringifiers (form, explode, space-delimited, pipe-delimited)
  - Tested format strategies with single values vs. collections
  - Confirmed backward compatibility with existing code

- Tested integration with mustache template rendering to ensure generated code works correctly

# Results
- Successfully implemented URL path template parsing with robustness against malformed inputs
- Created flexible query parameter parsing with support for all OpenAPI collection formats
- Implemented type-safe conversion between string values and target types using Jackson's type system
- Organized code with clear separation of concerns between stringification, destringification, and type conversion
- Ensured seamless integration with code generation templates to generate link builder methods

These utilities now enable HATEOAS compliance in the SDK by allowing clients to follow links received in responses. The implementation is type-safe, well-tested, and aligned with REST principles.

### CODE REVIEW

**Review by mohnoor94 — APPROVED** [review]

**Review by anssari1 — APPROVED** [review]

**Review by dtayeh — APPROVED** [review]

**Review by dtayeh — APPROVED** [review]

File: expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+            val expected =
+                mapOf(
+                    "param1" to mutableListOf("value1"),
+                    "param2" to mutableListOf("value2,value3")
+                )
+            assertEquals(expected, result)
+        }
+
+        @Test
+        fun `does not destringify query parameters with unknown strategy`() {
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt]: shouldn't we throw exception instead?

File: expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            // Then
+            assertEquals(emptyMap<String, String>(), result)
+        }
+    }
+
+    @Nested
+    inner class ParseUrlQueryParamsFromQueryTest {
+        @Test
+        fun `parses query string with pipe-delimited values using pipe strategy`() {
+            val query = "key=value1|value2|value3"
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt]: Do you see a value in adding one more `key` so we can cover multiple keys scenarios in these tests?
```suggestion
            val query = "key1=value1|value2|value3&key2=value1,|value2"
```

File: expediagroup-sdk-openapi-plugin/src/main/resources/templates/operations/params/link_builder_method.mustache [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                pathParams["{{{baseName}}}"].takeIf { it.isNotBlank() }?.let {
+                    this.{{{paramName}}} = {{{jacksonObjectMapper}}}.convertValue(it, jacksonTypeRef<{{>partials/datatype}}>())
+                }
+            }
+        {{/pathParams}}
+    }
+{{/hasPathParams}}
+
+{{#hasQueryParams}}
+    val destringifyStrategies = mapOf<String, String>(
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-openapi-plugin/src/main/resources/templates/operations/params/link_builder_method.mustache:24]: Is it expected to have more than one strategy in a single link?

File: expediagroup-sdk-openapi-plugin/src/main/kotlin/com/expediagroup/sdk/openapigenerator/util/OpenApiGeneratorConfigurator.kt [hunk 4]
```diff
@@ -153,7 +153,8 @@ object OpenApiGeneratorConfigurator {
         val defaultAdditionalProps =
             mapOf(
                 "namespace" to product.namespace,
-                "apiSuffix" to "Operation"
+                "apiSuffix" to "Operation",
+                "jacksonObjectMapper" to "${product.packageName}.configuration.OBJECT_MAPPER",
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-openapi-plugin/src/main/kotlin/com/expediagroup/sdk/openapigenerator/util/OpenApiGeneratorConfigurator.kt:157]: This assumes that all product SDKs have `OBJECT_MAPPER` in this specific package, I think it's fragile. Can we use standalone object mapper instead?

File: expediagroup-sdk-openapi-plugin/src/main/resources/templates/operation_params.mustache [hunk 5]
```diff
@@ -43,6 +43,10 @@ package com.expediagroup.sdk.{{namespace}}.operations
                 {{>operations/params/get_query_params}}
 
                 {{>operations/params/get_path_params}}
+
+                {{#link}}
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-openapi-plugin/src/main/resources/templates/operation_params.mustache:47]: Do we have custom logic that extracts the links? where do we get this `{{#link}}` from?

File: expediagroup-sdk-openapi-plugin/src/main/resources/templates/operations/params/link_builder_method.mustache (hunk 3 above)
Comment (Mohammad-Dwairi) [reply expediagroup-sdk-openapi-plugin/src/main/resources/templates/operations/params/link_builder_method.mustache:24]: should we move it inside `if (query != null)` below?

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt [hunk 6]
```diff
@@ -0,0 +1,107 @@
+package com.expediagroup.sdk.rest.util
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt:17]: let's add the license string please

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt [hunk 7]
```diff
@@ -32,53 +32,134 @@ fun interface StringifyQueryParam : (UrlQueryParam) -> String {
...
-        StringBuilder().apply {
-            append("${param.key}=")
-            append(param.value.joinToString(URLEncoder.encode("|", "UTF-8")))
-        }.toString()
+class UrlQueryParamDestringifiers {
+    companion object {
+        /**
+         * Destringifies a form-encoded query string into a list of values.
+         *
+         * @param query The form-encoded query string.
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: We don't have a param called `query` here, same for other functions

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt [hunk 8]
```diff
@@ -32,53 +32,134 @@ fun interface StringifyQueryParam : (UrlQueryParam) -> String {
...
+         * @return A list of values extracted from the query string.
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: I don't think this is what the function returns. Same for other functions

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt [hunk 9]
```diff
@@ -32,53 +32,134 @@ fun interface StringifyQueryParam : (UrlQueryParam) -> String {
...
-val stringifyPipeDelimited =
-    StringifyQueryParam { param ->
-        StringBuilder().apply {
-            append("${param.key}=")
-            append(param.value.joinToString(URLEncoder.encode("|", "UTF-8")))
-        }.toString()
+class UrlQueryParamDestringifiers {
+    companion object {
+        /**
+         * Destringifies a form-encoded query string into a list of values.
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]:
```suggestion
         * Parses a single - form-encoded - query param into its respective data type.
```
Can we also modify the rest of the functions below?

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt [hunk 10]
```diff
@@ -32,53 +32,134 @@ fun interface StringifyQueryParam : (UrlQueryParam) -> String {
...
+         *
+         * @param query The form-encoded query string.
+         * @return A list of values extracted from the query string.
+         */
+        val destringifyForm =
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: ***Optional:*** The IDE is complaining about "destringify" as a typo. Maybe we can update it to `deStringify`

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt [hunk 11]
```diff
@@ -94,8 +175,27 @@ val stringifyPipeDelimited =
 @Suppress("unused")
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: It's used in the unit tests now, can we remove this annotation?

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt [hunk 12]
```diff
@@ -32,53 +32,134 @@ fun interface StringifyQueryParam : (UrlQueryParam) -> String {
...
-    StringifyQueryParam { param ->
-        StringBuilder().apply {
-            append("${param.key}=")
-            append(param.value.joinToString(URLEncoder.encode(" ", "UTF-8")))
-        }.toString()
+        /**
+         * Converts a UrlQueryParam to an exploded form-encoded String.
+         * Example: key=value1&key=value2
+         */
+        val stringifyExplode =
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: **_Optional_**: Since you've moved these functions to `UrlQueryParamStringifiers` class, maybe it's good to drop the `stringify*` prefix from the functions names. so it becomes `UrlQueryParamStringifiers.explode()`

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt [hunk 13]
```diff
@@ -32,53 +32,134 @@ fun interface StringifyQueryParam : (UrlQueryParam) -> String {
...
-            append("${param.key}=")
-            append(param.value.joinToString(URLEncoder.encode(",", "UTF-8")))
-        }.toString()
-    }
+fun interface DestringifyQueryParam : (String, String) -> UrlQueryParam {
+    /**
+     * Converts the given String to a list of values.
+     *
+     * @param query the String to convert
+     * @return the list of values extracted from the String
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: These seems to be outdated, let's update them please

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt [hunk 14]
```diff
@@ -32,53 +32,134 @@ fun interface StringifyQueryParam : (UrlQueryParam) -> String {
...
- * Example: key=value1&key=value2
- */
-val stringifyExplode =
-    StringifyQueryParam { param ->
-        StringBuilder().apply {
-            append("${param.key}=")
-            append(param.value.joinToString("&${param.key}="))
-        }.toString()
-    }
+class UrlQueryParamStringifiers {
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]:
```suggestion
class UrlQueryParamStringifier {
```

Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: `class UrlQueryParamStringifiers` - Is there any reason this is not an `object`?

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+        require(templateSegments.size == pathSegments.size) {
+            "Template and path segments must have the same number of segments"
+        }
+    }.zipWithNext { templateSegment, pathSegment ->
+        templateSegment.zip(pathSegment)
+    }.flatten().filter { (key, value) ->
+        key != value && key.startsWith("{") && key.endsWith("}")
+    }.associate { (key, value) ->
+        key.removePrefix("{").removeSuffix("}") to value
+    }
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt]: Any concerns about this suggestions? I think it'a more readable 🤞🏻 

```kotlin
/**
 * Extracts the values of "{placeholders}" from a concrete URL path.
 *
 * Example:
 *   templateUrl = "/users/{id}/posts/{postId}"
 *   concretePath = "/users/42/posts/123"
 *   result = mapOf("id" to "42", "postId" to "123")
 *
 * @throws IllegalArgumentException if the two strings have different segment counts.
 */
fun extractPathParameters(templateUrl: String, concretePath: String): Map<String, String> {
    val templateParts = templateUrl.trim('/').split('/')
    val pathParts     = concretePath.trim('/').split('/')

    require(templateParts.size == pathParts.size) {
        "Template and path must contain the same number of segments"
    }

    return buildMap {
        templateParts
            .zip(pathParts)                                      
            .filter { (templatePart, pathPart) -> templatePart != pathPart }
            .filter { (templatePart, _) ->
                templatePart.startsWith('{') && templatePart.endsWith('}')
            }
            .forEach { (placeholder, value) ->
                put(placeholder.trim('{', '}'), value)            
            }
    }
}
```

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+                        is Collection<*> -> {
+                            addAll(value.map { it.toString() })
+                        }
+
+                        else -> {
+                            add(value.toString())
+                        }
+                    }
+                }
+            }
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt]: Any concerns about this suggestion? 

```kotlin
URLDecoder.decode(query, StandardCharsets.UTF_8.name())
            .splitToSequence('&')
            .filter { it.isNotEmpty() }
            .forEach { token ->
                val eq = token.indexOf('=')
                if (eq <= 0 || eq == token.lastIndex) return@forEach   // malformed

                val name = token.substring(0, eq)
                val rawValue = token.substring(eq + 1)

                val deStringifier = destringifyStrategies[name]
                    ?.let { swaggerCollectionFormatDestringifier[it] }

                val queryParamValues = deStringifier?.invoke(name, rawValue)?.value ?: listOf(rawValue)

                getOrPut(name, ::mutableListOf).addAll(queryParamValues)
            }
```

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+/**
+ * Parses URL path template parameters and returns a map of template segments to path segments.
+ *
+ * @param path The actual URL path.
+ * @param template The URL template with placeholders.
+ * @return A map where the keys are template segments and the values are the corresponding path segments.
+ * @throws IllegalArgumentException if either path or template is null.
+ */
+fun parseUrlPathTemplateParams(
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt]: If we move these functions to `UrlUtil` or `UrlHelper` class, we can simplify the name to
```suggestion
fun parsePathParams(
```

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+/**
+ * Parses URL query parameters from a query string and returns a map of parameter names to lists of values.
+ *
+ * @param query The URL query string.
+ * @param destringifyStrategies The strategy to destringify query parameter values (e.g., comma-delimited, space-delimited).
+ * @return A map where the keys are parameter names and the values are lists of parameter values.
+ */
+fun parseUrlQueryParamsFromQuery(
```
Comment (Mohammad-Dwairi) [inline expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt]: If we move these functions to `UrlUtil` or `UrlHelper` class, we can simplify the name to
```suggestion
fun parseQueryParams(
```

**Review by Mohammad-Dwairi — CHANGES_REQUESTED** [review]

File: expediagroup-sdk-openapi-plugin/src/main/kotlin/com/expediagroup/sdk/openapigenerator/util/OpenApiGeneratorConfigurator.kt (hunk 4 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-openapi-plugin/src/main/kotlin/com/expediagroup/sdk/openapigenerator/util/OpenApiGeneratorConfigurator.kt:157]: As discussed offline, creating multiple object mappers is costly and discouraged. Also, with the new pattern each product SDK maintains its own Object Mapper instance. Product SDKs can override the value assigned here through the generator `addtionalProperties` input.

File: expediagroup-sdk-openapi-plugin/src/main/resources/templates/operation_params.mustache (hunk 5 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-openapi-plugin/src/main/resources/templates/operation_params.mustache:47]: As discussed offline, this shall be maintained in the product SDK.

File: expediagroup-sdk-openapi-plugin/src/main/resources/templates/operations/params/link_builder_method.mustache (hunk 3 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-openapi-plugin/src/main/resources/templates/operations/params/link_builder_method.mustache:24]: > Is it expected to have more than one strategy in a single link?

Yes

Comment (OmarAlJarrah) [reply expediagroup-sdk-openapi-plugin/src/main/resources/templates/operations/params/link_builder_method.mustache:24]: > should we move it inside `if (query != null)` below?

It does not make a real difference I think. Don't you agree?

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt (hunk 13 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Done.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt (hunk 14 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Done.

Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Done.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt (hunk 12 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Done.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt (hunk 9 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Done.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt (hunk 7 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Done.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt (hunk 8 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Done.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt (hunk 10 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Offline dissc.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt (hunk 11 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt]: Done.

File: expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt (hunk 2 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt]: Done.

File: expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt (hunk 1 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt]: Done.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt (hunk 15 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt]: Suggestion accepted.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt (hunk 18 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt]: Done.

File: expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt (hunk 16 above)
Comment (OmarAlJarrah) [reply expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt]: Change accepted.

**Review by Mohammad-Dwairi — APPROVED** [review]
